recipes/boost/all/conanfile.py

```diff
@@ -304,11 +317,23 @@ def configure(self):
         elif self.options.shared:
             del self.options.fPIC
 
+        if self.options.i18n_backend != "deprecated":
+            self.output.warn("i18n_backend option is deprecated, do not use anymore.")
+            if self.options.i18n_backend == "iconv":
+                self.options.i18n_backend_iconv = True
+                self.options.i18n_backend_icu = False
+            if self.options.i18n_backend == "icu":
+                self.options.i18n_backend_iconv = False
+                self.options.i18n_backend_icu = True
+            if self.options.i18n_backend == "None":
+                self.options.i18n_backend_iconv = False
+                self.options.i18n_backend_icu = False
         if self.options.without_locale:
-            self.options.i18n_backend = None
+            self.options.i18n_backend_iconv = False
+            self.options.i18n_backend_icu = False
```

Comment:
When an option dos not make sense, you can remove it `del self.options.i18n_backend_iconv`.
You can access it by using `self.options.get_safe("i18n_backend_iconv")`.